File: app/discover/scan_aggregate.rb
#!/usr/bin/env ruby

# scan a host aggregate for hosts

require 'singleton'
require_relative 'db_fetch_aggregate_hosts'
require_relative 'scan_host'

class ScanAggregate < Scanner
  include Singleton
  
  def initialize()
    super([
      {:fetcher => DbFetchAggregateHosts.new(), :children_scanner => ScanHost.instance}
    ])
  end
  
end

File: app/discover/scan_aggregate.py
# scan a host aggregate for hosts

from singleton import Singleton
from scanner import Scanner

class ScanAggregate(Scanner, metaclass=Singleton):
  
  def __init__(self):
    super(ScanAggregate, self).__init__([
      {
        "type": "host_ref",
        "fetcher": "DbFetchAggregateHosts"
      }
    ])
